### PR TITLE
fix: dynamic default model selection based on configured providers

### PR DESCRIPTION
**Pull request by joshkotrous**
### What does this PR do?

Fixes the issue where starting Apex without a saved model preference would always default to an Anthropic model (the first entry in `AVAILABLE_MODELS`), causing a "No Anthropic API key" error when Anthropic wasn't the configured provider.

**Root cause:** `AgentProvider` initialized its model state to `AVAILABLE_MODELS[0]` (an Anthropic model) and only ran an async `useEffect` to correct it. If no providers were configured yet (e.g. first launch flow), or if a non-Anthropic provider was set up, the initial Anthropic model could be used before the effect ran or the effect would bail out early, leaving the stale Anthropic default in place.

**Changes:**

1. **New `getDefaultModelForConfig()` utility** (`src/core/providers/utils.ts`) — centralized logic that dynamically selects the best default model based on which providers are configured:
   - Pensar → Claude Opus 4.1 (Pensar gateway)
   - Anthropic → Claude Opus 4.1
   - OpenAI → GPT-4o
   - Google → Gemini 2.5 Flash
   - OpenRouter → Claude Opus 4.6
   - Bedrock → first available model
   - Falls back to the first available model for any configured provider

2. **`AgentProvider`** (`src/tui/context/agent.tsx`) — now reads from the config context (reactive to provider setup changes) instead of a one-shot async config read. Initial model state is computed synchronously from the current config, and re-evaluated whenever config changes (e.g. after completing provider onboarding).

3. **`ConfigView`** (`src/tui/components/chat/config-view.tsx`) — uses the same dynamic default instead of hard-coding an Anthropic model.

4. **CLI** (`src/cli.ts`) — `pentest` and `targeted-pentest` commands now load config to determine the default model dynamically when `--model` is not specified.

### How did you verify your code works?

- All existing tests pass (428 passed, 15 skipped integration tests)
- Added 12 new unit tests for `getDefaultModelForConfig` covering all provider combinations, priority ordering, and edge cases
- TypeScript type check passes (`bun run tsc`)
- Lint passes (`bun run lint`)
- Format check passes on changed files

<div><a href="https://cursor.com/agents/bc-99a7a1ce-8cc7-4885-be51-13babccc3063"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-web-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-web-light.png">Open in Web</picture></a>&nbsp;<a href="https://cursor.com/background-agent?bcId=bc-99a7a1ce-8cc7-4885-be51-13babccc3063"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-cursor-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-cursor-light.png">Open in Cursor</picture></a>&nbsp;</div>

### CODE REVIEW

**Comment by joshkotrous** [comment]
[src/core/providers/utils.ts](https://github.com/pensarai/apex/pull/402/files/b3580f67f489b21e2a6cd39db6b4e8831c968c7b#diff-2901901aa808d147eafbb4eb74bb6f1a90fcf29dc635cfd4acc285003ebc9a4d)
];

const PREFERRED_MODEL_BY_PROVIDER: Record<string, string> = {
  pensar: "pensar:anthropic.claude-opus-4-1-20250805-v1:0",

@cursor it should default to the pensar opus-4-6 model

File: src/core/providers/utils.ts [hunk 1]
```diff
@@ -7,6 +7,23 @@ import {
...
+const PROVIDER_PREFERENCE_ORDER: ProviderType[] = [
+  "pensar",
+  "anthropic",
+  "openai",
+  "google",
+  "openrouter",
+  "bedrock",
+];
+
+const PREFERRED_MODEL_BY_PROVIDER: Record<string, string> = {
```
Comment (jorgeraad) [inline src/core/providers/utils.ts:19]: any reason we aren't defaulting to the best models (i.e., opus 4.6, codex 5.4, gemini 3.1 pro)?

File: src/tui/components/chat/config-view.tsx [hunk 2]
```diff
@@ -32,24 +35,28 @@ export function ConfigView({ config, onBack, onStart }: ConfigViewProps) {
...
-  const [selectedModel, setSelectedModel] = useState<ModelInfo>({
-    id: "claude-sonnet-4-20250514",
-    name: "Claude Sonnet 4",
-    provider: "anthropic",
+  const [selectedModel, setSelectedModel] = useState<ModelInfo>(() => {
+    if (config) {
+      const dynamicDefault = getDefaultModelForConfig(config);
+      if (dynamicDefault) return dynamicDefault;
+    }
+    return {
```
Comment (jorgeraad) [inline src/tui/components/chat/config-view.tsx:43]: We can change `getDefaultModelForConfig` to fall back to `PREFERRED_MODEL_BY_PROVIDER['anthropic]` instead of null, return `string` and remove this.

**Review by jorgeraad — APPROVED** [review]
looks good, left two nit comments.
